framework/include/multiapps/MultiApp.h

```diff
@@ -555,6 +578,9 @@ class MultiApp : public MooseObject,
   const PerfID _backup_timer;
   const PerfID _restore_timer;
   const PerfID _reset_timer;
+
+  // MooseApp for setAppOutputFileBase
+  friend class MooseApp;
```

Comment:
This was the easy way out to avoid having `setAppOutputFileBase` be public. I can possibly do a similar "key" strategy that we do to get write access in `ReporterData`. I will let @loganharbour comment on this.

Reply:
We decided to just make the method public. I'm fine with this - I don't think it's overly concerning.